### PR TITLE
Update OpenAI-based factual consistency to use the whole generated output

### PR DESCRIPTION
**Pull request by yosukehigashi**
For the OpenAI-based factual consistency metric, we've anecdotally seen that the metric works better when comparing the full generated output with the source text, as opposed to comparing individual sentences of the generated output with the source text and aggregating the scores. A major additional benefit of this is that it makes the generated explanations a lot easier to understand.

On the English factual consistency benchmarks, we see that the new formulation improves correlation for the XSUM dataset, but regresses correlation for the CNN dataset.

| Data | Per-Sentence Scoring | Full Output Scoring |
|-------------|---------|---------|
| **QAGS-XSUM**    | Pearson correlation = 0.40<br>Spearman correlation = 0.40<br>Kendall-Tau correlation = 0.38 | Pearson correlation = 0.49<br>Spearman correlation = 0.50<br>Kendall-Tau correlation = 0.48  |
| **QAGS-CNN**    | Pearson correlation = 0.67<br>Spearman correlation = 0.59<br>Kendall-Tau correlation = 0.52 | Pearson correlation = 0.50<br>Spearman correlation = 0.45<br>Kendall-Tau correlation = 0.42  |

Basically no change on the Japanese JNLI factual consistency benchmark

| Data | Per-Sentence Scoring | Full Output Scoring |
|-------------|---------|---------|
| **JNLI**    | Pearson correlation = 0.546<br>Spearman correlation = 0.556<br>Kendall-Tau correlation = 0.518 | Pearson correlation = 0.549<br>Spearman correlation = 0.562<br>Kendall-Tau correlation = 0.521  |

### CODE REVIEW

**Comment by kennysong** [comment]
Interesting! So it's not a clear win in metric quality, but we get much better explanations and faster performance, so the change makes sense.

**Review by kennysong — APPROVED** [review]
